ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -26,33 +49,38 @@ void RSkComponentImage::OnPaint(
   auto component = getComponentData();
   auto const &imageProps =
       *std::static_pointer_cast<ImageProps const>(component.props);
-  if (imageProps.sources.empty()) {
-    return;
+  sk_sp<SkImage> imageData = nullptr;
+  if (!imageProps.sources.empty() && imageProps.sources[0].type == ImageSource::Type::Local ) {
+    imageData=getLocalImage(imageProps.sources[0]);
   }
-  const auto source = imageProps.sources[0];
-
-  if (source.type == ImageSource::Type::Local && !source.uri.empty()) {
-    assert(source.uri.substr(0, 14) == "file://assets/");
-    std::string path = "./" + source.uri.substr(7);
-
+  auto imageEventEmitter = std::static_pointer_cast<ImageEventEmitter const>(component.eventEmitter);
+  if(imageData) {
+/* Emitting Load completed Event*/
+    imageEventEmitter->onLoad();
     Rect frame = component.layoutMetrics.frame;
-    SkRect rect = SkRect::MakeXYWH(frame.origin.x, frame.origin.y, frame.size.width, frame.size.height);
+    SkRect frameRect = SkRect::MakeXYWH(frame.origin.x, frame.origin.y, frame.size.width, frame.size.height);
     auto const &imageBorderMetrics=imageProps.resolveBorderMetrics(component.layoutMetrics);
-
- /* Draw order 1. Background 2. Image 3. Border*/
+    SkRect targetRect = computeTargetRect({imageData->width(),imageData->height()},frameRect,imageProps.resizeMode);
+    SkPaint paint;
+/* TO DO: Handle filter quality based of configuration. Setting Low Filter Quality as default for now*/
+    paint.setFilterQuality(DEFAULT_IMAGE_FILTER_QUALITY);
+    if(imageProps.resizeMode == ImageResizeMode::Repeat){
+      sk_sp<SkImageFilter> imageFilter(SkImageFilters::Tile(targetRect,frameRect ,nullptr));
+      paint.setImageFilter(std::move(imageFilter));
+    }
+/* Draw order 1. Background 2. Image 3. Border*/
     drawBackground(canvas,frame,imageBorderMetrics,imageProps.backgroundColor,imageProps.opacity);
-    RNS_PROFILE_START(drawImage)
-    sk_sp<SkImage> imageData=getImageData(path.c_str());
-    if(imageData) {
-      canvas->drawImageRect(imageData, rect, nullptr);
-    } else {
-      RNS_LOG_ERROR("Draw Image Failed for:" << path);
+    canvas->save();
+      /*TO DO: Handle filter quality based of configuration. Setting Low Filter Quality as default for now*/
```

Comment:
Do you need `filter quality` comment here also?

Reply:
Comment here was a duplicate. Missed to notice. Will remove .